### ISSUE TITLE
Build fails with custom prefix

### ISSUE THREAD

**Issue by iamazeem** [liquidaty/zsv#10]
**Description**

I was trying to build with a custom prefix i.e. `build`:

```shell
./configure --prefix=build
make install
```

The `lib` builds successfully but `cli` fails while linking with these errors:

```text
/usr/bin/ld: cannot find -lzsv
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:275: ~/zsv/build/Linux/rel/cc/bin/cli] Error 1
make[1]: Leaving directory '~/zsv/app'
make: *** [Makefile:46: install] Error 2
```

Also, the sequence of linker flags seemed incorrect:

```shell
cc -pipe -ffunction-sections -fdata-sections ... -lzsv -Lbuild/lib -lpthread -fwhole-program -march=native -ldl
```

`-L` should precede `-l`.
And, `-Lbuild/bin` doesn't seem right.
Shouldn't this be complete path from the build directory i.e.:

```shell
$ tree build
build/
└── Linux
    └── rel
        └── cc
            ├── bin
            ├── external
            │   ├── ...
            ├── lib
            │   └── libzsv.a
            └── objs
                ├── cli_2json.o
                ├── ...
                ├── utils
                │   ├── arg.o
                │   ├── ...
                │   └── writer.o
                └── zsv.o
```

**Expected**

The `cli` should build successfully.

**Actual**

The `cli` is failing with linking error as `libzsv.a` is not found.

**Comment by liquidaty** [comment]
Should be fixed now for e61e231d12bf0b9139d95232866b07de6efa9268 and subsequent commits. Thank you!

**Comment by liquidaty** [comment]
-L / -l order addressed in ef1ad67128d18a471298d6ac3323f8c452315606

**Comment by iamazeem** [comment]
@liquidaty : Thank you for your quick response! :+1: The `prefix` issue is fixed now.
